go/libraries/doltcore/sqle/enginetest/branch_control_test.go

```diff
@@ -971,15 +966,10 @@ func TestBranchControl(t *testing.T) {
 				if host == "" {
 					host = "localhost"
 				}
-				var ctx *sql.Context
-				var ok bool
-				if ctx, ok = ctxMap[user+"@"+host]; !ok {
-					ctx = enginetest.NewContextWithClient(harness, sql.Client{
-						User:    user,
-						Address: host,
-					})
-					ctxMap[user+"@"+host] = ctx
-				}
+				ctx = ctx.NewCtxWithClient(sql.Client{
```

Comment:
i dont think contexts are reusable, or at least we make that hard assumption across enginetests. not sure i would have expected the old way to work

Reply:
We definitely don't expect them to be reusable, but this at least gets the tests to pass. The old way (reusing contexts) got some other tests to pass that simply instantiating a new context every assertion failed to do. I wouldn't expect this workaround to work for other engine tests.